tool-testsuite/test/org/antlr/v4/test/tool/TestSymbolIssues.java

```diff
@@ -542,4 +542,22 @@ public void testLabelsForTokensWithMixedTypesLRWithoutLabels() {
 
 		testErrors(test, false);
 	}
+
+	// ISSUE: https://github.com/antlr/antlr4/issues/2788
+	@Test public void testUndefinedLabel() {
+		String[] test = {
+				"grammar Test;" +
+				"root\n" +
+				"    : root a\n" +
+				"    | b [error]\n" +
+				"    ;\n" +
+				"\n" +
+				"a: 'a';\n" +
+				"b: 'b';",
+
+				"error(" + ErrorType.INTERNAL_ERROR.code + "): Test.g4:2:30: internal error: Rule error undefined \n"
+		};
+
+		testErrors(test, false);
+	}
```

Comment:
I know this is already more than 2 years old, but I'm currently on porting this test and I wonder how it can even work. Neither the old ANTLRParser.g4, nor the new ANTLRv4Parser.g4 grammar allow an arg action block after a rule ref (with or w/o element options). Consequently I get a syntax error for that grammar code. Yet the test seems to indicate the grammar is valid and only errors out because a non-existing rule is referenced. Can someone please explain, how that's possible?